libs/s25main/CheatCommandTracker.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "CheatCommandTracker.h"
+#include "Cheats.h"
+#include "driver/KeyEvent.h"
+
+namespace {
+auto makeCircularBuffer(const std::string& str)
+{
+    return boost::circular_buffer<char>{cbegin(str), cend(str)};
+}
+const auto cheatStr = makeCircularBuffer("winter");
```

Comment:
I'll change the name but I'd rather keep this as a circular_buffer as well. Even though it is not necessary it lets me compare and get the length easily. Ideally 'winter' would just be some type of string, but then a comparison operator/function would be necessary (and in case of a constexpr C string an additional strlen-type call). Such code would then belong in some helpers somewhere and should be tested, etc.